node/src/components/console/tasks.rs

```diff
@@ -0,0 +1,335 @@
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Json => |data: &EraDump| {
+                let mut buf = serde_json::to_vec(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as JSON: {}", err))
+                })?;
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Bincode => |data: &EraDump| {
+                bincode::serialize(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as bincode: {}", err))
+                })
+            },
+        }
+    }
+
+    /// Processes a single command line sent from a client.
+    async fn process_line<REv>(
+        &mut self,
+        effect_builder: EffectBuilder<REv>,
+        writer: &mut OwnedWriteHalf,
+        line: &str,
+    ) -> io::Result<()>
+    where
+        REv: From<DumpConsensusStateRequest> + Send,
+    {
+        debug!(%line, "line received");
+        match Command::from_line(line) {
+            Ok(ref cmd) => {
+                info!(?cmd, "processing command");
+                match cmd.action {
+                    Action::Session => {
+                        self.send_outcome(writer, &Outcome::success("showing session info"))
+                            .await?;
+                        self.send_to_client(writer, &self).await?;
+                    }
+                    Action::Set { quiet, output } => {
+                        let mut changed = false;
+
+                        if let Some(quiet) = quiet {
+                            self.quiet = quiet;
+                            changed = true;
+                        }
+
+                        if let Some(output) = output {
+                            self.output = output;
+                            changed = true;
+                        }
+
+                        if changed {
+                            self.send_outcome(writer, &Outcome::success("session updated"))
+                                .await?;
+                        } else {
+                            self.send_outcome(writer, &Outcome::success("session unchanged"))
+                                .await?;
+                        }
+                    }
+                    Action::DumpConsensus { era } => {
+                        let output = effect_builder
+                            .console_dump_consensus_state(
+                                era.map(EraId::new),
+                                self.create_era_dump_serializer(),
+                            )
+                            .await;
+
+                        match output {
+                            Ok(ref data) => {
+                                self.send_outcome(
+                                    writer,
+                                    &Outcome::success("dumping consensus state"),
+                                )
+                                .await?;
```

Comment:
IMO this entry should be right at the beginning of this clause and the error should be logged regardless - i.e. it looks to me that these first `send_outcome`s calls are indicating an action being taken regardless of the results later.

Reply:
I'm not sure I understand this. To be clear, we are sending the outcome, which is the status of the consensus dumping operation first, then the actual data.

We cannot send down any errors encountered in `write_all` if that's what you mean, as we cannot send the client a sending error!

Reply:
I meant that the approach here is that `Console` will send the first response that is a confirmation of an action being taken (ignored if `quiet = true`) and then the result of taking the action. `"dumping consensus state"` is the former and should be sent regardless of the actual outcome.

Reply:
It just sometimes happens to be sent early, but these are not intended as confirmations of reception, but rather outcomes.